lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -3163,7 +3163,8 @@ static bool usePrespecialized(
 
         if (lowered.isBuiltinBridgeObject() && layout->isBridgeObject()) {
           newSubs.push_back(genericParam->getASTContext().TheBridgeObjectType);
-        } else if (lowered.hasRetainablePointerRepresentation()) {
+        } else if (lowered.hasRetainablePointerRepresentation() &&
```

Comment:
Doesn’t this mean that hasRetainablePointerRepresentation() should be changed? I’m pretty sure other callers also assume AnyObject compatibility when this returns true.